### ISSUE TITLE
Use `sanitize_textarea_field`?

### ISSUE THREAD

**Issue by kraftner** [WebDevStudios/CMB2#846]
### Expected Behavior:

As of [WP 4.7](https://core.trac.wordpress.org/changeset/38944) `sanitize_text_field` has a sibling function `sanitize_textarea_field` which would make sense as a default for CMB2 for `textarea`, `textarea_small` and maybe also `wysiwyg` and `oembed`.

### Actual Behavior:

CMB2 uses [`wp_kses_post`](https://github.com/WebDevStudios/CMB2/blob/6dacea9c1af974bd3ce253b7ff74403c23d74b8e/includes/CMB2_Sanitize.php#L379).

### Possible solutions:

One *could* change the default, but I am not sure this would be a great idea as this is a pretty big change. But what about at least mentioning that in [the Wiki](https://github.com/WebDevStudios/CMB2/wiki/Field-Parameters#sanitization_cb)? Something along the lines of:

> Since Version 4.7. WordPress has a sanitization function for textareas: `sanitize_textarea_field()`. So similiar to the default `sanitize_text_field()` CMB2 uses for text fields you could use that for textareas:
> ```
> /**
>  * Sanitizes using the core sanitize_textarea_field() function
>  */
> function your_prefix_sanitize_textarea( $content ) {
>     return sanitize_text_field( $content );
> }
> ```

I'm ready to make a PR, just asking for opinions here first.

**Comment by tw2113** [comment]
May be best left to the wiki for the moment, CMB2 still has a required version down at 3.8. Using it in the plugin itself would necessitate bumping that up quite a lot. Not to say people shouldn't be updating their sites :D 

I'll let Justin chime in as well

**Comment by kraftner** [comment]
Yeah, that is what I thought too. I mainly wanted to achieve two things here:

- Discuss if that generally is a safe (in the sense of a security-focused) approach to go with and what the advantages and disadvantages would be.
I already noticed that `sanitize_textarea_field()`has the advantage of leaving special chars like `&` as they are, while `wp_kses_post`shines in the handling of HTML (not stripping allowed html tags out completely, keeping the content of disallowed tags).
- If and how this should be covered in the Wiki.

**Comment by kraftner** [comment]
Since a couple of years later I hit the same issue today I am bumping this to maybe find an answer to my two questions above this time. :smile:

**Comment by tw2113** [comment]
@jtsternberg thoughts on any parts of this? Perhaps it's time to bump our minimum WP version at least?
